common4j/src/main/com/microsoft/identity/common/java/telemetry/relay/AbstractTelemetryRelayClient.java

```diff
@@ -39,33 +36,33 @@
 public abstract class AbstractTelemetryRelayClient<T> implements ITelemetryObserver<T> {
 
     private static final String TAG = AbstractTelemetryRelayClient.class.getSimpleName();
-    // Filters to apply on the events captured.
-    // This basically performs an AND operation on the filters to determine whether the event will be relayed.
-    private final List<ITelemetryEventFilter<T>> eventFilters = new ArrayList<>();
+    // Filter to apply on the events captured.
+    private ITelemetryEventFilter<T> mEventFilter = null;
 
     @Override
     public synchronized void onReceived(T telemetryData) {
         final String methodTag = TAG + ":onReceived";
 
-        for (final ITelemetryEventFilter<T> filter : eventFilters) {
-            // Prevent event relay if any of the filters returns false.
-            if (!filter.shouldRelay(telemetryData)) {
-                return;
-            }
+        T filteredData = telemetryData;
+
+        if (mEventFilter != null) {
+            filteredData = mEventFilter.apply(telemetryData);
         }
 
-        try {
-            relayEvent(telemetryData);
-        } catch (TelemetryRelayException e) {
-            Logger.error(methodTag, "Error relaying telemetry data", e);
+        if (filteredData != null) {
+            try {
+                relayEvent(filteredData);
+            } catch (TelemetryRelayException e) {
+                Logger.error(methodTag, "Error relaying telemetry data", e);
```

Comment:
What might cause a relay exception?

Reply:
I'm thinking in this case we may want to hold onto the data in some sort of telemetry cache and make another attempt at relaying it in the future....just something to think about...we don't have to solve that problem right now